### DISCUSSION TITLE
PlaySounds with simultaneous calls

### DISCUSSION THREAD

**Discussion by pedroalpenna** [sipsorcery-org/sipsorcery#443] [Q&A]
Hello, 

It is possible to use the PlaySounds from the SIPExamples to make simultaneous calls? The idea is to use it to call, for example, to five devices and play the same sound from each answered device. For example:

- User 1 answered from device 1: play sound.
- User 3 answered from device 3: play same sound from beginning.

We used that reference:

https://github.com/sipsorcery-org/sipsorcery/issues/237

We also used the "await Task.WhenAll" and "Parallel.ForEach", but what we have achieved so far is to ring the devices but only one device play the sound when answered. If another device is answered nothing happens.

Best regards.

**Comment by sipsorcery** [comment]
> play same sound from beginning.

Doesn't that just mean each caller is independent and gets the same sound played from the start?

There's nothing special for that case. Each new incoming call should be answered with a new `SIPUserAgent` on a dedicated thread per call.

If you want to play the same audio stream at the same position for all callers that's a lot trickier. The audio stream would need to be forked and sent to all incoming callers simultaneously. It's possible but there are no existing examples for that.
